### PR TITLE
I fixed a segfault problem and installing has been simplified

### PR DESCRIPTION
**Pull request by SystemDisc**
I fixed a segfault problem when setting 'FLASH_FULLSCREEN_DISPLAY' that I stumbled over when testing things. stdlib.h should have been included.

But, more importantly I modified the makefile. I hope you appreciate it.

### CODE REVIEW

File: fullscreenhack.c [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 #include <dlfcn.h>
 #include <stdio.h>
+#include <stdlib.h>
```
Comment (SystemDisc) [inline fullscreenhack.c:6]: this is needed for 'getenv'

File: fullscreenhack.c [hunk 2]
```diff
@@ -37,26 +38,26 @@ int choose_screen(Display *display, XineramaScreenInfo *screens,
     Window root = RootWindow(display, DefaultScreen(display));
     int x, y, wx, wy, n;
     unsigned int mask;
-    Bool result;
```
Comment (SystemDisc) [inline fullscreenhack.c:40]: this wasn't used; it was only set to 'XQueryPointer's return value

File: fullscreenhack.c [hunk 3]
```diff
@@ -89,6 +90,8 @@ Status XGetGeometry(Display *display, Drawable d, Window *root_return,
 
         *width_return = screens[n].width;
         *height_return = screens[n].height;
+	fprintf(stderr, "\nUsing: screen[%d] %dx%d+%d+%d\n\n", n, screens[n].width, // output selected resolution to stderr
```
Comment (SystemDisc) [inline fullscreenhack.c:93]: I think this is more informative than "match found"

File: Makefile [hunk 4]
```diff
@@ -19,5 +19,30 @@ clean:
 	rm -f *.o *.so tester
 
 install:
-	cp libfullscreenhack.so /usr/lib/nspluginwrapper/i386/linux/
-	cp npviewer /usr/lib/nspluginwrapper/noarch/
+	@echo "'install' is not a valid option. Read the README file!"
```
Comment (SystemDisc) [inline Makefile:22]: I added this so people would be forced to read the README for install options

File: Makefile [hunk 5]
```diff
@@ -19,5 +19,30 @@ clean:
...
+
+simpleinstall:
```
Comment (SystemDisc) [inline Makefile:24]: just installs the '.so'; people will have to LD_PRELOAD themselves

File: Makefile [hunk 6]
```diff
@@ -19,5 +19,30 @@ clean:
...
+ifeq ($(BITS),64)
+	cp libfullscreenhack.so /usr/lib64/libfullscreenhack.so
+else
+	cp libfullscreenhack.so /usr/lib/libfullscreenhack.so
+endif
+
+
+firefox:
```
Comment (SystemDisc) [inline Makefile:32]: this option will move /usr/bin/firefox to /usr/bin/firefox.main and have /usr/bin/firefox run /usr/bin/firefox.main with LD_PRELOAD

File: README [hunk 7]
```diff
@@ -24,45 +25,94 @@ The monitor sizes sent are now based on Xinerama.
...
-For native Flash plugin:
-------------------------
+After a successful install:
 
-Build it:
+/usr/bin/firefox is moved to /usr/bin/firefox.main if it hasn't been moved already
+and /usr/bin/firefox contains code to preload the hack while running /usr/bin/firefox.main
+This part is not important to most end-users
+
+Other-browsers:
```
Comment (SystemDisc) [inline README:89]: I want to add specific support for browsers other than firefox

File: Makefile (hunk 5 above)
Comment (SystemDisc) [inline Makefile:24]: this is the safest option

File: fullscreenhack.c (hunk 1 above)
Comment (SystemDisc) [inline fullscreenhack.c:6]: without this included gcc somehow manages to compile but when running with 'FLASH_FULLSCREEN_DISPLAY' set to anything a segfault happens

File: README (hunk 7 above)
Comment (SystemDisc) [inline README:89]: right now 'make simpleinstall' should be run for other browsers and 'LD_PRELOAD' set by the user

**Comment by ali1234** [comment]
Thanks. I will pull the bug fixes soon. I am not sure about the installer stuff though. I think the right way to do it is to create .desktop launchers for each browser. They can go in the system wide config or the user's homedir, and it doesn't need to overwrite any packaged files. They are easy to make too. You just copy the existing .desktop and change eg "firefox %u" to "env LD_PRELOAD=/home/al/Source/fullscreenhack/libfullscreenhack.so firefox %u"

**Comment by SystemDisc** [comment]
I think creating .desktop launchers is an excellent idea! However, do all linux desktop environments use .desktop launchers? If not, neither way is entirely flawless. I do agree that overwriting packaged files is not the best option; it's the only thing I could think of. As a side note, your makefile overwrites packaged files of nspluginwrapper.
Also, thanks for being so prompt about responding :D

**Comment by ali1234** [comment]
I've pulled the fixes. Closing this issue. Please see issue #7 for installation stuff.

**Pull request closed without merging**